README.md

```diff
@@ -16,35 +16,37 @@ Heimdall is inspired by the Zero Trust idea and also by [Pomerium](https://www.p
 
-It can do so 
-* standalone as a proxy in front of your service or web server that rejects unauthorized requests and forwards authorized ones to your end points, or 
-* integrated into any other proxy, ingress controller or API gateway, like Kong, NGNIX, Envoy, Traefik, Contour, Ambassador and many more. Here that other proxy will forward the incoming request to heimdall and depending on its response either forward the original request, verified and updated by heimdall to your upstream service, or reject it with the information provided by heimdall.
+It can do so:
 
-In both cases is acts as a Policy Enforcement and to some degree a Policy Decision Point according to  [NIST Zero Trust Architecture (SP 800-207)](https://doi.org/10.6028/NIST.SP.800-207)
+* Standalone as a proxy in front of your service or web server that rejects unauthorized requests and forwards authorized ones to your end points, or 
+* Integrated into any other proxy, ingress controller or API gateway, like Kong, NGNIX, Envoy, Traefik, Contour, Ambassador and many more. Here that other proxy will forward the incoming request to heimdall and depending on its response either forward the original request, verified and updated by heimdall to your upstream service, or reject it with the information provided by heimdall.
 
-## How does authentication, authorization and transformation happen
+In both cases it acts as a Policy Enforcement and to some degree a Policy Decision Point according to [NIST Zero Trust Architecture (SP 800-207)](https://doi.org/10.6028/NIST.SP.800-207)
 
-The above said decision and transformation process happens via rules, respectively rule sets, which can be controlled by each and every upstream service individually and loaded by heimdall from different sources, like
+## How does authentication, authorization and transformation work
+
+The decision-making and transformation processes in Heimdall are governed by rules or respectively rule sets. These rule sets can be independently configured and managed by each upstream service. Heimdall dynamically loads these rules from a variety of sources, including:
 
-* the local file system,
-* any HTTP endpoint,
-* cloud storages, like AWS S3, Google's GC, etc. and
 * `RuleSet` kubernetes resources (a corresponding CRD is shipped with the helm chart)
+* Cloud storages, like AWS S3, Google's GC, etc.
+* Local file system
+* Any HTTP endpoint
 
-That way, these rule sets can not only be managed centrally, but be deployed together with each particular upstream service as well without the need to restart or redeploy heimdall. Indeed, these rule sets are optional first class citizens of the upstream service and allow:
+That way, these rule sets cannot only be managed centrally, but be deployed together with each particular upstream service as well. Without the need to restart or redeploy heimdall. Indeed, these rule sets are optional first class citizens of the upstream service and allow:
```

Comment:
> ... with each particular upstream service as well. Without the need to restart or redeploy heimdall.
The dot before "without" slightly disrupts the reading flow. While I appreciate the effort to make the sentences more concise, it's important to ensure the original meaning remains clear. In this case, besides stating that these rules can be deployed alongside the upstream service, it's also essential to convey that this configuration can be applied without downtime, as heimdall does not need to be restarted for the rules to take effect. I fear this information is now lost.